lib/providers/httpTransaction/index.js

```diff
@@ -82,12 +82,32 @@ HttpTransaction.prototype._initTrace = function (type, data) {
     service: this.service
   }
 
-  // only save to span for server events
-  if (['sr', 'ss'].indexOf(type) > -1) {
-    dataTrace.span = data.url
-    dataTrace.host = data.host
+  dataTrace.span = data.url
+  dataTrace.host = data.host
+
+  // fallback to appName if service id is not present
+  if (isNaN(this.service)) {
+    dataTrace.serviceName = this.serviceName
   }
 
+  this.partials[traceId] = defaults(this.partials[traceId] || {}, dataTrace, {
+    events: []
+  })
+
+  return this.partials[traceId]
+}
+
+HttpTransaction.prototype.closeSpan = function (data) {
+  var traceId = data.id || ''
+
+  var dataTrace = {
+    trace: traceId,
+    service: this.service
+  }
+
+  dataTrace.span = data.url
```

Comment:
these were already added at open. Is it possible that were missing at open?
Should we have `statusCode` here?